cloudinit/sources/DataSourceWSL.py

```diff
@@ -238,22 +327,64 @@ def check_instance_id(self, sys_cfg) -> bool:
 
     def _get_data(self) -> bool:
         self.vendordata_raw = None
-        seed_dir = cloud_init_data_dir()
+        user_home = find_home()
+        seed_dir = cloud_init_data_dir(user_home)
+        user_data: Optional[Union[dict, bytes]] = None
+        requires_multipart = False
 
         try:
             self.metadata = load_instance_metadata(
                 seed_dir, self.instance_name
             )
-            file = self.find_user_data_file(seed_dir)
-            self.userdata_raw = cast(
-                str, util.load_binary_file(file.as_posix())
-            )
-            return True
+        except (ValueError, IOError) as err:
+            LOG.error("Unable to load metadata: %s", str(err))
+
+        try:
+            agent_data = load_agent_data(user_home)
+            user_data = load_landscape_data(self.instance_name, user_home)
```

Comment:
If we generate any kind of exception from these 2 functions, we will not try to process and fallback to valid WSL user-data below as we will have ejected from this try block. Is this desired behavior?

It seems load_yaml_or_bin function is resilient enough to return None on FileNotFound or non-YAML binaries. So, maybe we don't expect this to raise unless there is a file permissions error. If so we'd fall through without any user_data defined as _get_data would return False skipping this DataSourceWSL detection.